### ISSUE TITLE
Feature request: Preconfigured PlatformIO support

### ISSUE THREAD

**Issue by Domush** [grblHAL/core#92]
I'm coming from Marlin, which compiles in VS Code out of the box, and have no experience using C compilers beyond following the occasional how-to. 

Is there anyone who can create and push a pre-configured platformio.ini for this project with support for the most popular boards? 

I've read the Wiki here on how to compile with PlatformIO but it is super vague, and skips a lot of steps and board-specific items that only those who don't need a Wiki to begin with know of. Making a Wiki how-to meant for pros kind of defeats the purpose of a Wiki how-to.

Is there someone who can add platformio support for those of us who don't live and bleed C, but just want to build a CNC?

For reference, I'm trying to build for an SKR 1.3 board, but I'm hoping a robust platformio.ini implementation won't need to be tweaked per-board, like how the Marlin implementation works: you just change the board ID and hit compile.

Thanks.

**Comment by dresco** [comment]
Hi, the PlatformIO section in the core wiki is pretty out of date (was carried over from the old repository structure). 

Have you looked at the readme in the STM32F4 repository? That's a much better starting point..

**Comment by dresco** [comment]
@terjeio Do you want to delete the "Using PlatformIO for Arduino development" page from the core Wiki? 

My build instructions in there are outdated now, and I think the subject is much better covered by the README.md files in both STM32F4xx and iMXRT1062 driver repositories...

**Comment by terjeio** [comment]
> Do you want to delete the "Using PlatformIO for Arduino development" page from the core Wiki?

Perhaps it should be updated or replaced with (links to?) some generic info?

**Comment by conqueror433** [comment]
I totally agree with @dresco It will be very helpful if procedure of compiling can be simplified.

@dresco Have you had success compiling Grbl-HAL to SKR 1.3?  I am also looking to do the same for my laser engraver.

**Comment by dresco** [comment]
> @dresco Have you had success compiling Grbl-HAL to SKR 1.3?

Am assuming you meant this for @Domush, rather than me? Afraid I don't have any LPC boards, else I'd give it a go!

**Comment by conqueror433** [comment]
Ahh yess. Sorry for that.

**Comment by tcurdt** [comment]
platformio support would be super great.

**Comment by terjeio** [comment]
A webui for bulding binaries is now underway - check this [discussion and preview](https://github.com/grblHAL/core/discussions/203).

**Comment by terjeio** [comment]
The WebBuilder can be used to [output a .ini-file for platformio](https://github.com/grblHAL/core/wiki/Compiling-grblHAL#platformio) for most drivers, except the RP2040, that can be used/modified for local builds.

**Comment by tcurdt** [comment]
> except the RP2040

Just out of curiosity: What's so special about the RP2040 that it is singled out?

**Comment by terjeio** [comment]
> What's so special about the RP2040 that it is singled out?

Could be that platformio for the RP2040 was not available at the time I set up the Web Builder? Cannot remember now. Anyway compilation runs in a Docker container, and it works, so why change it?

**Comment by tcurdt** [comment]
> so why change it?

As said: I was just curious about the reason.

FWIW: This works for my pio projects on the RP2040

```
[env:pico]
platform = raspberrypi
board = pico
framework = arduino
```
